library/bigip_software.py

```diff
@@ -46,64 +40,148 @@
-    description:
-      - The path to an optional Hotfix to install. This parameter requires that
-        the C(software) parameter be specified.
-    required: false
-    default: None
-    aliases:
-      - hotfix_image
   reuse_inactive_volume:
     description:
-      - Automatically chooses the first inactive volume in alphanumeric order. If there
-        is no inactive volume, new volume with incremented volume name will be created.
-        For example, if HD1.1 is currently active and no other volume exists, then the
-        module will create HD1.2 and install the software. If volume name does not end with
-        numeric character, then add .1 to the current active volume name. When C(volume)
-        is specified, this option will be ignored.
+      - Automatically chooses the first inactive volume in alphanumeric order. 
+        If there is no inactive volume, new volume with incremented volume name 
+        will be created. For example, if HD1.1 is currently active and no other 
+        volume exists, then the module will create HD1.2 and install the 
+        software. If volume name does not end with numeric character, 
+        then add .1 to the current active volume name. When C(volume) is 
+        specified, this option will be ignored.
     required: false
   state:
     description:
-      - When C(installed), ensures that the software is uploaded and installed,
-        on the system. The device is not, however, rebooted into the new software.
-        When C(activated), ensures that the software is uploaded, installed, and
-        the system is rebooted to the new software. When C(present), ensures
-        that the software is uploaded. When C(absent), only the uploaded image
-        will be removed from the system
+      - When C(installed), ensures that the software is uploaded/downloaded 
+        and installed on the system. The device is not, however, 
+        rebooted into the new software.
+      - When C(activated), ensures that the software is uploaded/downloaded, 
+        installed, and the system is rebooted to the new software. 
+      - When C(present), ensures that the software is uploaded/downloaded. 
+      - When C(absent), only the uploaded/downloaded image 
+        will be removed from the system.
     required: false
     default: activated
     choices:
       - absent
       - activated
       - installed
       - present
+    volume:
+    description:
+      - The volume to install the software and, optionally, the hotfix to. This
+        parameter is only required when the C(state) is either C(activated) or
+        C(installed).
+    required: false
   software:
     description:
       - The path to the software (base image) to install. The parameter must be
         provided if the C(state) is either C(installed) or C(activated).
+      - If C(remote_src) is used, this parameter will have to be an 
+        C(unauthenticated) C(HTTP) or C(HTTPS)link to the software image.
```

Comment:
no you wouldnt, the url can just include those. you dont do any special parsing of the url